addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -2450,9 +2450,20 @@ export function enforceWhitespace(el, offset, direction, rule) {
     }
 }
 
-export function rgbToHex(rgb = '') {
+export function rgbToHex(rgb = '', bg = [255, 255, 255]) {
     if (rgb.startsWith('#')) {
         return rgb;
+    } else if (rgb.startsWith('rgba')) {
```

Comment:
Isn't it a little weird that the first parameter is string but not the other while both are colors ? It's like you'd have a function like `sum(a, b)` where `a` is a string and `b` is a number. I feel like both parameters should be of same type, whether it be string or number[], and that the function itself does the necessary conversion if needed. If I were you I'd have it take two strings and do the trim with default `0` inside the function here so that you don't have to worry about the string colors you get from the DOM when calling this function.

Reply:
You're right. Two strings make more sense. Formatted like the css values I guess then simply: `rgbToHex('rgba(0, 0, 0, 0)', 'rgba(0, 0, 0, 0)')`?
I didn't rename the function because you know, api in stable etc. but it does sting a little lol

Reply:
@dmo-odoo Actually that made me realize we come into trouble if the background color itself is rgba because then we need to compute its "flattened" value by getting the background color of its parent. So I suggest to pass the node instead of the background color.

```js
/**
 * Takes a color (rgb, rgba or hex) and returns its hex representation. If the
 * color is given in rgba, the background color of the node whose color we're
 * converting is used in conjunction with the alpha to compute the resulting
 * color (using the formula: `alpha*color + (1 - alpha)*background` for each
 * channel).
 *
 * @param {string} rgb
 * @param {HTMLElement} [node]
 * @returns {string} hexadecimal color (#RRGGBB)
 */
export function rgbToHex(rgb = '', node = null) {
    if (rgb.startsWith('#')) {
        return rgb;
    } else if (rgb.startsWith('rgba')) {
        const values = rgb.match(/[\d\.]{1,5}/g) || [];
        const alpha = parseFloat(values.pop());
        // Retrieve the background color.
        let bgRgbValues = [];
        if (node) {
            let bgColor = getComputedStyle(node).backgroundColor;
            if (bgColor.startsWith('rgba')) {
                // The background color is itself rgba so we need to compute
                // the resulting color using the background color of its
                // parent.
                bgColor = rgbToHex(bgColor, node.parentElement);
            }
            if (bgColor && bgColor.startsWith('#')) {
                bgRgbValues = (rgbBackgroundColor.match(/[\da-f]{2}/gi) || []).map(val => parseInt(val, 16));
            } else if (bgColor && bgColor.startsWith('rgb')) {
                bgRgbValues = (rgbBackgroundColor.match(/[\d\.]{1,5}/g) || []).map(val => parseInt(val));
            }
        }
        bgRgbValues = bgRgbValues.length ? bgRgbValues : [255, 255, 255]; // Default to white.

        return (
            '#' +
            values.map((value, index) => {
                const converted = Math.floor(alpha * parseInt(value) + (1 - alpha) * bgRgbValues[index]);
                const hex = parseInt(converted).toString(16);
                return hex.length === 1 ? '0' + hex : hex;
            }).join('')
        );
    } else {
        return (
            '#' +
            (rgb.match(/\d{1,3}/g) || [])
                .map(x => {
                    x = parseInt(x).toString(16);
                    return x.length === 1 ? '0' + x : x;
                })
                .join('')
        );
    }
}
```